### PR TITLE
Implemented logic for part SQL and part JDBC from HW16

### PR DESCRIPTION
**Pull request by andriydovbenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <description>Simple SQL and JDBC educational project</description>
+
+    <groupId>com.cursor</groupId>
+    <artifactId>JDBCAndSQLOperation</artifactId>
```
Comment (YStakhniak) [inline pom.xml]: artifactId Should be lowercase. Words should be separated by "-".

File: src/main/java/com/execution/Executor.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
+package com.execution;
+
+import com.util.DbUtil;
+import com.model.Account;
+import com.model.Profile;
+
+public class Executor {
+    private static final int ROW_NUMBER = 500;
+    private final Account accounts = new Account();
```
Comment (YStakhniak) [inline src/main/java/com/execution/Executor.java]: accounts or account?

File: src/main/java/com/execution/Executor.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    private final Profile profiles = new Profile();
```
Comment (YStakhniak) [inline src/main/java/com/execution/Executor.java]: profiles or profile?

File: src/main/java/com/execution/Executor.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        printAccountsAndProfiles();
+
+        removeAccountAndProfile(vasilAccount, vasilProfile);
+        printAccountsAndProfiles();
+
+        getAccountAndProfileFromTables(ROW_NUMBER);
+        tryToPrintAccountAndProfileInformation(accountFromTable, profileFromTable);
+    }
+
+    private void tryToPrintAccountAndProfileInformation(Account account, Profile profile) {
```
Comment (YStakhniak) [inline src/main/java/com/execution/Executor.java]: try to print? Why "try?"

File: src/main/java/com/model/Account.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package com.model;
+
+import lombok.*;
+
+@Getter
```
Comment (YStakhniak) [inline src/main/java/com/model/Account.java]: You can use @Data instead @Getter @Setter

File: src/main/java/com/model/Account.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+@ToString
+public class Account implements Model{
+    private @NonNull int ID;
+    private @NonNull String USER_NAME;
+    private String FIRST_NAME;
+    private String LAST_NAME;
```
Comment (YStakhniak) [inline src/main/java/com/model/Account.java]: Why all these fields have names like constants?

File: src/main/java/com/model/Model.java [hunk 7]
```diff
@@ -0,0 +1,4 @@
+package com.model;
+
+public interface Model {
```
Comment (YStakhniak) [inline src/main/java/com/model/Model.java]: Why do you need an empty interface?

File: src/main/java/com/model/Profile.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+@EqualsAndHashCode
+@ToString
+public class Profile implements Model {
+    private @NonNull int ID;
+    private @NonNull String USER_NAME;
```
Comment (YStakhniak) [inline src/main/java/com/model/Profile.java]: The same as above

File: src/main/java/com/util/DbUtil.java [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+
+import java.sql.*;
+
+public class DbUtil {
+    private static final String URL = "jdbc:postgresql://localhost:5432/dev_profiles_db";
+    private static final String USERNAME = "postgres";
+    private static final String PASSWORD = "1607";
+    private static final String ACCOUNTS_TABLE = "dev_profiles_db.public.accounts";
+    private static final String PROFILES_TABLE = "dev_profiles_db.public.profiles";
+    private static DbUtil INSTANCE;
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Why this field is also like constant?

File: src/main/java/com/util/DbUtil.java [hunk 10]
```diff
@@ -0,0 +1,216 @@
...
+
+    private DbUtil() {
+    }
+
+    public static DbUtil getINSTANCE() {
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: getINSTANCE????

File: src/main/java/com/util/DbUtil.java [hunk 11]
```diff
@@ -0,0 +1,216 @@
...
+    public void insertModelIntoDB(Model model) {
+        if (model instanceof Account) {
+            insertAccountIntoDB((Account) model);
+        } else {
+            insertProfileIntoDB((Profile) model);
+        }
+    }
+
+    private void insertAccountIntoDB(Account model) {
+        String SQL = "INSERT INTO " + ACCOUNTS_TABLE +
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Who told you to use upper case for local variables?

File: src/main/java/com/util/DbUtil.java [hunk 12]
```diff
@@ -0,0 +1,216 @@
...
+                "(id, first_name, last_name, city, gender, username) VALUES(?,?,?,?,?,?)";
+        String[] accountData = {model.getFIRST_NAME(), model.getLAST_NAME(),
+                model.getCITY(), model.getGENDER(), model.getUSER_NAME(),
+                String.valueOf(model.getID())};
+        tryToInsertIntoDB(accountData, SQL);
+    }
+
+    private void insertProfileIntoDB(Profile model) {
+        String SQL = "INSERT INTO " + PROFILES_TABLE +
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Same as above

File: src/main/java/com/util/DbUtil.java [hunk 13]
```diff
@@ -0,0 +1,216 @@
...
+                "(id, username, job_title, department, company, skill) VALUES(?,?,?,?,?,?)";
+        String[] profileData = {model.getUSER_NAME(), model.getJOB_TITLE(),
+                model.getDEPARTMENT(), model.getCOMPANY(), model.getSKILL(),
+                String.valueOf(model.getID())};
+        tryToInsertIntoDB(profileData, SQL);
+    }
+
+    private void tryToInsertIntoDB(String[] data, String SQLCommand) {
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Not SQLCommand but sqlQuery or just query.

File: src/main/java/com/util/DbUtil.java [hunk 14]
```diff
@@ -0,0 +1,216 @@
...
+            }
+            preparedStatement.executeUpdate();
+        } catch (SQLException | NumberFormatException ex) {
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    public void printTableIntoConsole(Model model) {
+        if (model instanceof Account) {
+            String SQL = "SELECT * FROM " + ACCOUNTS_TABLE;
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Same as above

File: src/main/java/com/util/DbUtil.java [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+            tryToCreateConnectionAndPrintTable(SQL);
+        } else {
+            String SQL = "SELECT * FROM " + PROFILES_TABLE;
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Same as above

File: src/main/java/com/util/DbUtil.java [hunk 16]
```diff
@@ -0,0 +1,216 @@
...
+                        resultSet.getString(5) + "," + resultSet.getString(6));
+            }
+        } catch (SQLException ex) {
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    public Model selectDataFromTable(int numberOfRow, Model model) {
+        if (model instanceof Account) {
+            String SQL = "SELECT id, first_name, last_name, city, gender, username FROM " + ACCOUNTS_TABLE +
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Same as above

File: src/main/java/com/util/DbUtil.java [hunk 17]
```diff
@@ -0,0 +1,216 @@
...
+            int id = ((Account) model).getID();
+            tryToUpdateDB(SQL, id);
+        } else {
+            String SQL = getSQLForUpdatingProfile((Profile) model);
+            int id = ((Profile) model).getID();
+            tryToUpdateDB(SQL, id);
+        }
+    }
+
+    private void tryToUpdateDB(String SQL, int id) {
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Why "try"?

File: src/main/java/com/util/DbUtil.java [hunk 18]
```diff
@@ -0,0 +1,216 @@
...
+            int id = ((Account) model).getID();
+            tryToRemoveFromTableById(SQL, id);
+        } else {
+            String SQL = "DELETE FROM " + PROFILES_TABLE + " WHERE id = ?";
+            int id = ((Profile) model).getID();
+            tryToRemoveFromTableById(SQL, id);
+        }
+    }
+
+    private void tryToRemoveFromTableById(String SQL, int id) {
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: Same as above

File: src/main/java/com/util/DbUtil.java [hunk 19]
```diff
@@ -0,0 +1,216 @@
...
+    }
+
+    public static DbUtil getINSTANCE() {
+        if (INSTANCE == null) {
+            INSTANCE = new DbUtil();
+        }
+        return INSTANCE;
+    }
+
+    private Connection connect() throws SQLException {
```
Comment (YStakhniak) [inline src/main/java/com/util/DbUtil.java]: establishConnection something like that

File: pom.xml (hunk 1 above)
Comment (andriydovbenko) [reply pom.xml]: > artifactId Should be lowercase. Words should be separated by "-".

done

File: src/main/java/com/execution/Executor.java (hunk 2 above)
Comment (andriydovbenko) [reply src/main/java/com/execution/Executor.java]: > accounts or account?

corrected

File: src/main/java/com/execution/Executor.java (hunk 3 above)
Comment (andriydovbenko) [reply src/main/java/com/execution/Executor.java]: > profiles or profile?

corrected

File: src/main/java/com/execution/Executor.java (hunk 4 above)
Comment (andriydovbenko) [reply src/main/java/com/execution/Executor.java]: > try to print? Why "try?"

I wasn't sure how to name the method correctly when the result could be different from what you expect

fixed

File: src/main/java/com/model/Account.java (hunk 5 above)
Comment (andriydovbenko) [reply src/main/java/com/model/Account.java]: > You can use @DaTa instead @getter @Setter

done

File: src/main/java/com/util/DbUtil.java (hunk 15 above)
Comment (andriydovbenko) [reply src/main/java/com/util/DbUtil.java]: > Same as above

done

File: src/main/java/com/util/DbUtil.java (hunk 16 above)
Comment (andriydovbenko) [reply src/main/java/com/util/DbUtil.java]: > Same as above

fixed

File: src/main/java/com/util/DbUtil.java (hunk 17 above)
Comment (andriydovbenko) [reply src/main/java/com/util/DbUtil.java]: > Why "try"?

just a mistake

File: src/main/java/com/util/DbUtil.java (hunk 18 above)
Comment (andriydovbenko) [reply src/main/java/com/util/DbUtil.java]: > Same as above

corrected

File: src/main/java/com/util/DbUtil.java (hunk 19 above)
Comment (andriydovbenko) [reply src/main/java/com/util/DbUtil.java]: > establishConnection something like that

method name was fixed

**Pull request merged**